### DISCUSSION TITLE
SmartDate to text

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#1617] [General]
I have an interesting scenario, one of our datetime fields in sql database allows null (long story :)),  if the value is 'empty' datetime according to SmartDate, we want to show the user  text "Not Available". We are declaring our property as of type SmartDate in both Csla business object and our Data transfer object.
I know I could declare the property  of type string  but I was wondering if we could use something different?
Kind regards

**Comment by rockfordlhotka** [comment]
It depends on your UI technology. Some UI technologies have controls that allow you to put default text in the control if the underlying property value is empty.

**Comment by brinawebb** [comment]
We use nullable dates in a lot of our applications and they show a blank string when there's no date. 

Your situation may dictate something different, but in general, personally, from a UX standpoint, I think a blank string is better than "Not Available". Your end-users will be very comfortable with this concept as most apps like Excel that deal with a lot of data do this. I'm assuming this is in a data entry form? On a grid, I would for sure leave it blank as it's much easier for the eye to see that there's something vs. nothing there.
